### PR TITLE
tdb: update to @1.4.6

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

`tdb` has been forgotten. Update been requested here: https://trac.macports.org/ticket/45436
However there was an issue reported in the trac ticket, which may not have been fixed. Could someone take a look, please?

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.7
Xcode 12.4

macOS 10.6.8 Server
Xcode 3.2.6

macOS 10.6 PPC (10A190)
Xcode 3.2

### CODE REVIEW

**Comment by barracuda156** [comment]
Hmm, the bot fails on this:
```
./configure: line 20: /opt/local/bin/python3.9: No such file or directory
```
How come it has no `python3.9`?

File: databases/tdb/Portfile [hunk 1]
```diff
@@ -1,32 +1,41 @@
...
                     writers from trampling on each other. TDB is also \
                     extremely small.
 
 homepage            http://sourceforge.net/projects/tdb/
-master_sites        sourceforge:project/tdb/tdb/${version}
-checksums           md5 6b643fdeb48304010dcd5f675e458b58 \
-                    sha1 d1876522f1b8ffa8cf844a1f6605e0c32d387a7a \
-                    rmd160 5163caf1cc76c9f3a69e88f7438ecd98ec5e52cf
+master_sites        https://www.samba.org/ftp/tdb/
+checksums           md5     10046f5068c4b25c5beabaf0378b0289 \
```
Comment (reneeotten) [inline databases/tdb/Portfile:21]: please remove oudated checksums`md5` and `sha1`

**Review by reneeotten — CHANGES_REQUESTED** [review]

**Comment by reneeotten** [comment]
> How come it has no `python3.9`?

because you don't declare a dependency on it?

**Comment by ryandesign** [comment]
> However there was an issue reported in the trac ticket, which may not have been fixed. Could someone take a look, please?

You can see by trying to run one of the installed programs that the problem reported in the ticket still exists:

```
$ tdbbackup
dyld: Library not loaded: /opt/local/var/macports/build/_Users_rschmidt_macports_macports-ports-ryandesign-fork_databases_tdb/tdb/work/tdb-1.4.6/bin/default/libtdb.dylib
  Referenced from: /opt/local/bin/tdbbackup
  Reason: image not found
Abort trap: 6
```

It's unfortunate that the developers of this software appear to have zero interest in addressing even the most basic of Mac-related build issues.

I have some further fixups to make to this port and will see if I can add a workaround for this problem.

**Comment by reneeotten** [comment]
let's close this PR as it hasn't had any activity for more than a year now, and wait for @ryandesign to make the changes he mentions here

**Pull request closed without merging**